src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1193,6 +1197,87 @@ protected override void HandlePauseRequestAsync(IRequestResponder<PauseArguments
             m_program.CauseBreak();
             responder.SetResponse(new PauseResponse());
         }
+        
+        protected override void HandleGotoRequestAsync(IRequestResponder<GotoArguments> responder)
+        {
+            var response = new GotoResponse();
+            if (!m_isStopped)
+            {
+                responder.SetResponse(response);
+                return;
+            }
+
+            var gotoTarget = m_gotoCodeContexts[responder.Arguments.TargetId];
+            IDebugThread2 thread = null;
+            lock (m_threads)
+            {
+                if (!m_threads.TryGetValue(responder.Arguments.ThreadId, out thread))
+                    throw new AD7Exception("Could not find thread!");
+            }
+            BeforeContinue();
+            var builder = new ErrorBuilder(() => AD7Resources.Error_UnableToSetNextStatement);
+            try
+            {
+                builder.CheckHR(thread.SetNextStatement(null, gotoTarget));
+            }
+            catch (AD7Exception e)
+            {
+                m_isStopped = true;
+                responder.SetError(new ProtocolException(e.Message));
+            }
+
+            responder.SetResponse(response);
+        }
+        
+        protected override void HandleGotoTargetsRequestAsync(IRequestResponder<GotoTargetsArguments, GotoTargetsResponse> responder)
+        {
+            var response = new GotoTargetsResponse();
+
+            var source = responder.Arguments.Source;
+            // TODO: handle this for disassembly debugging
```

Comment:
>// TODO: handle this for disassembly debugging [](start = 12, length = 46)

I feel like the right want to handle disassembly is to add a `gotoInstruction` request which would take a `instructionReference` and `offset` (like "InstructionBreakpoint")